lib/lwm2m_carrier/include/lwm2m_carrier.h

```diff
@@ -98,4 +193,181 @@ const char *lwm2m_carrier_timezone(int32_t tz_offset, int32_t dst);
 void lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event);
 /**@} */
 
+/**
+ * @brief      Set the available power sources supported and used by the LWM2M device.
+ *
+ * @note       It is necessary to call this function before any other device power source related
+ *             functions listed in this file, as any updates of voltage/current measurements performed
+ *             on power sources that have not been reported will be discarded.
+ * @note       Upon consecutive calls of this function, the corresponding current and voltage measurements
+ *             will be reset to 0. Similarly, the battery status will be set to UNKNOWN and the battery
+ *             level to 0%.
+ *
+ * @param[in]  power_sources          Array of available device power sources.
+ * @param[in]  power_source_count     Number of power sources currently used by the device.
+ *
+ * @return     -E2BIG    If the reported number of power sources is bigger than the maximum supported.
```

Comment:
I'm pretty sure that `retval` should be used in case return values are listed like this.

Reply:
true